plugin/storage/kafka/options.go

```diff
@@ -33,19 +35,41 @@ const (
 	// EncodingZipkinThrift is used for spans encoded as Zipkin Thrift.
 	EncodingZipkinThrift = "zipkin-thrift"
 
-	configPrefix          = "kafka.producer"
-	suffixBrokers         = ".brokers"
-	suffixTopic           = ".topic"
-	suffixProtocolVersion = ".protocol-version"
-	suffixEncoding        = ".encoding"
-	defaultBroker         = "127.0.0.1:9092"
-	defaultTopic          = "jaeger-spans"
-	defaultEncoding       = EncodingProto
+	configPrefix           = "kafka.producer"
+	suffixBrokers          = ".brokers"
+	suffixTopic            = ".topic"
+	suffixEncoding         = ".encoding"
+	suffixRequiredAcks     = ".required-acks"
+	suffixCompression      = ".compression"
+	suffixCompressionLevel = ".compression.level"
+
+	defaultBroker           = "127.0.0.1:9092"
+	defaultTopic            = "jaeger-spans"
+	defaultEncoding         = EncodingProto
+	defaultRequiredAcks     = "local"
+	defaultCompression      = "none"
+	defaultCompressionLevel = 0
+	suffixProtocolVersion   = ".protocol-version"
 )
 
 var (
 	// AllEncodings is a list of all supported encodings.
 	AllEncodings = []string{EncodingJSON, EncodingProto, EncodingZipkinThrift}
+
+	//CompressionModes is a mapping of supported CompressionType to compressionCodec
+	CompressionModes = map[string]sarama.CompressionCodec{
+		"none":   sarama.CompressionNone,
+		"gzip":   sarama.CompressionGZIP,
+		"lz4":    sarama.CompressionLZ4,
+		"snappy": sarama.CompressionSnappy,
```

Comment:
I would recommend this instead, so that you don't have to `switch` later for default compression level
```go
var compressionModes = map[string]struct {
	compressor   sarama.CompressionCodec
	defaultLevel int
}{
	"none": {
		compressor:   sarama.CompressionNone,
		defaultLevel: 0,
	},
	"gzip": {
		compressor:   sarama.CompressionGZIP,
		defaultLevel: 6,
	},
}
```

or even extend it with min/max